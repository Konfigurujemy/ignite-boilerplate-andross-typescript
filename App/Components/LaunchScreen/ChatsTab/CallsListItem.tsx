import * as React from "react";
import {ListItem, Left, Body, Right, Thumbnail, Text} from 'native-base';
import Fonts from "../../../Themes/Fonts";
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {View} from "react-native";
import {connect} from "react-redux";
import {colorScheme} from "../../../Themes/Colors";

interface CallsListItemProps {
  avatar: string,
  name: string,
  state: number,
  time: string,
  isDarkMode?: boolean
}


interface callData {
  text: string,
  color: string,
  icon: string,
}

function callStateDataChooser (state: number) : callData {
  switch (state){
    case 0 : return {
      text: 'No response',
      color: 'red',
      icon: 'call-missed'
    };
    case 1: return {
      text: 'Successful',
      color: 'green',
      icon: 'call-made'
    };
    case 2: return {
      text: 'Rejected',
      color: 'red',
      icon: 'call-missed-outgoing'
    };
    default: return {
      text: 'Unknown',
      color: 'red',
      icon: 'call-missed-outgoing'
    }
  }
}


const CallListItem = (item : CallsListItemProps) => {

  const callStateData = callStateDataChooser(item.state);
  const ColorScheme = colorScheme(item.isDarkMode);

  return (
    <ListItem button avatar>
      <Left>
        <Thumbnail source={{uri: item.avatar}}/>
      </Left>
      <Body  style={{alignItems:'flex-start'}}>
      <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.fullToneText}}>{item.name}</Text>
      <View style = {{flexDirection:'row'}}>
        <MaterialIcons
          name={callStateData.icon}
          size={28}
          style={{marginBottom: -3}}
          color={ callStateData.color}
        />
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.midToneText}} note>{callStateData.text}</Text>
      </View>
      </Body>
      <Right>
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.lightText}} note>{item.time}</Text>
      </Right>
    </ListItem>
);
};

const mapStateToProps = state => {
  return {
    isDarkMode: state.isDarkMode.isDarkMode
  };
};

export default connect(mapStateToProps)(CallListItem);
