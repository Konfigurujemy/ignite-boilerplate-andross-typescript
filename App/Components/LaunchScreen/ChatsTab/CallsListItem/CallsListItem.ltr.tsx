import * as React from "react";
import {ListItem, Left, Body, Right, Thumbnail, Text} from 'native-base';
import Fonts from "../../../../Themes/Fonts";
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {View} from "react-native";
import {connect} from "react-redux";
import CallStateToData from "../../../../Transforms/CallStateToData";
import {CallsListItemProps} from "./index";

const CallListItem = (item : CallsListItemProps) => {

  const callStateData = CallStateToData(item.state);
  const ColorScheme = item.colorScheme;

  return (
    <ListItem button avatar>
      <Left>
        <Thumbnail source={{uri: item.avatar}}/>
      </Left>
      <Body  style={{alignItems:'flex-start'}}>
      <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.fullToneText}}>{item.name}</Text>
      <View style = {{flexDirection:'row'}}>
        <MaterialIcons
          name={callStateData.icon}
          size={28}
          style={{marginBottom: -3}}
          color={ callStateData.color}
        />
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.midToneText}} note>{callStateData.text}</Text>
      </View>
      </Body>
      <Right>
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.lightText}} note>{item.time}</Text>
      </Right>
    </ListItem>
);
};

const mapStateToProps = state => {
  return {
    isDarkMode: state.appSettings.isDarkMode,
    colorScheme: state.appSettings.colorScheme
  };
};

export default connect(mapStateToProps)(CallListItem);
