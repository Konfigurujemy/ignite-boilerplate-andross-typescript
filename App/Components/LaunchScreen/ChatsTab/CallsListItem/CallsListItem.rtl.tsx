import * as React from "react";
import {ListItem, Left, Body, Right, Thumbnail, Text} from 'native-base';
import Fonts from "../../../../Themes/Fonts";
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {View} from "react-native";
import {connect} from "react-redux";
import CallStateToData from "../../../../Transforms/CallStateToData";

interface CallsListItemProps {
  avatar: string,
  name: string,
  state: number,
  time: string,
  isDarkMode?: boolean
}

const CallListItem = (item : CallsListItemProps) => {

  const callStateData = CallStateToData(item.state);
  const ColorScheme = item.colorScheme;

  return (
    <ListItem button avatar>
      <Left>
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.lightText}} note>{item.time}</Text>
      </Left>
      <Body  style={{alignItems:'flex-start'}}>
      <Text style ={{alignSelf:'flex-end',fontFamily:Fonts.type.farsi,color:ColorScheme.fullToneText}}>{item.name}</Text>
      <View style = {{alignSelf:'flex-end',flexDirection:'row'}}>
        <MaterialIcons
          name={callStateData.icon}
          size={28}
          style={{marginBottom: -3}}
          color={ callStateData.color}
        />
        <Text style ={{fontFamily:Fonts.type.farsi,color:ColorScheme.midToneText}} note>{callStateData.text}</Text>
      </View>
      </Body>
      <Right>
        <Thumbnail source={{uri: item.avatar}}/>
      </Right>
    </ListItem>
  );
};

const mapStateToProps = state => {
  return {
    isDarkMode: state.appSettings.isDarkMode,
    colorScheme: state.appSettings.colorScheme
  };
};

export default connect(mapStateToProps)(CallListItem);
