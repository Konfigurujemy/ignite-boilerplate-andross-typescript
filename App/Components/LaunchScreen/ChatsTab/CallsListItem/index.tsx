import * as React from "react";
import Rtl from './CallsListItem.rtl';
import Ltr from './CallsListItem.ltr';
import {connect} from "react-redux";
import {ColorScheme} from "../../../../Themes/Colors";


export interface CallsListItemProps {
  avatar: string,
  name: string,
  state: number,
  time: string,
  isDarkMode?: boolean,
  isLtr?:boolean,
  colorScheme?:ColorScheme
}

const Direction = (props:CallsListItemProps) => props.isLtr? <Ltr {...props} /> : <Rtl {...props} />;

const mapStateToProps = state => {
  return {
    isLtr: state.appSettings.isLtr
  };
};

export default connect(mapStateToProps)(Direction);
