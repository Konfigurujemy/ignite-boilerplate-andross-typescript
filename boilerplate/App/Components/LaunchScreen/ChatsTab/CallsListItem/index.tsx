import * as React from "react";
import Rtl from './CallsListItem.rtl';
import Ltr from './CallsListItem.ltr';
import {connect} from "react-redux";
import {ColorScheme} from "../../../../Themes/Colors";


export interface CallsListItemProps {
  avatar: string,
  name: string,
  state: number,
  time: string,
  isDarkMode?: boolean,
  isRtl?:boolean,
  colorScheme?:ColorScheme
}

export const Direction = (props:CallsListItemProps) => props.isRtl? <Rtl {...props} /> : <Ltr {...props} />;

const mapStateToProps = state => {
  return {
    isRtl: state.appSettings.isRtl
  };
};

export default connect(mapStateToProps)(Direction);
