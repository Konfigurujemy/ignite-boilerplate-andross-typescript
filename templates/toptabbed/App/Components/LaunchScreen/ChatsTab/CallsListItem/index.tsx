import * as React from "react";
import Rtl from './CallsListItem.rtl';
import Ltr from './CallsListItem.ltr';
import {ColorScheme} from "../../../../Themes/Colors";


export interface CallsListItemProps {
  avatar: string,
  name: string,
  state: number,
  time: string,
  isDarkMode?: boolean,
  isRtl?:boolean,
  colorScheme?:ColorScheme
}

export default (props:CallsListItemProps) => props.isRtl? <Rtl {...props} /> : <Ltr {...props} />;
