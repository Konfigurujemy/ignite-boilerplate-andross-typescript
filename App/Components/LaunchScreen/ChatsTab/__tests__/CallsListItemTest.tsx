import "react-native";
import * as React from "react";
import {Direction as CallsListItem} from "../CallsListItem";
import * as renderer from 'react-test-renderer';
import {colorScheme} from "../../../../Themes/Colors";
import createStore from "../../../../Redux";
import {Provider} from 'react-redux'

test("Renders Correctly" , () => {
  const store = createStore();

  const tree = renderer.create(<Provider store={store}><CallsListItem
      avatar={'http://digitalspyuk.cdnds.net/17/25/980x490/landscape-1498216547-avatar-neytiri.jpg'}
      name={'sajjad'}
      state={2}
      time={'3:30'}
      colorScheme={colorScheme(false)}
      isLtr={true}
      isDarkMode={false}
  /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
