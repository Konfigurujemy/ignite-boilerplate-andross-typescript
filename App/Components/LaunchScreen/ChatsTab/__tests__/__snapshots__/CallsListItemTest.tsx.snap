// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    button={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flex": 0,
          "flexDirection": "row",
          "paddingTop": 14,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://digitalspyuk.cdnds.net/17/25/980x490/landscape-1498216547-avatar-neytiri.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          },
          Object {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            },
            Object {
              "color": "#212121",
              "fontFamily": "BYekan+",
            },
          ]
        }
        uppercase={false}
      >
        sajjad
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "red",
                "fontSize": 28,
              },
              Object {
                "marginBottom": -3,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ?
        </Text>
        <Text
          note={true}
          style={
            Array [
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "200",
                "marginLeft": null,
                "marginRight": 10,
              },
              Object {
                "color": "#424242",
                "fontFamily": "BYekan+",
              },
            ]
          }
          uppercase={false}
        >
          Rejected
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        note={true}
        style={
          Array [
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            },
            Object {
              "color": "#9E9E9E",
              "fontFamily": "BYekan+",
            },
          ]
        }
        uppercase={false}
      >
        3:30
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    button={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flex": 0,
          "flexDirection": "row",
          "paddingTop": 14,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://digitalspyuk.cdnds.net/17/25/980x490/landscape-1498216547-avatar-neytiri.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          },
          Object {
            "alignItems": "flex-start",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            },
            Object {
              "color": "#212121",
              "fontFamily": "BYekan+",
            },
          ]
        }
        uppercase={false}
      >
        sajjad
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "red",
                "fontSize": 28,
              },
              Object {
                "marginBottom": -3,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ?
        </Text>
        <Text
          note={true}
          style={
            Array [
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "200",
                "marginLeft": null,
                "marginRight": 10,
              },
              Object {
                "color": "#424242",
                "fontFamily": "BYekan+",
              },
            ]
          }
          uppercase={false}
        >
          Rejected
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        note={true}
        style={
          Array [
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            },
            Object {
              "color": "#9E9E9E",
              "fontFamily": "BYekan+",
            },
          ]
        }
        uppercase={false}
      >
        3:30
      </Text>
    </View>
  </View>
</View>
`;
